src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -2,46 +2,251 @@
+
+            allFieldsResponse = await MakeAzureInstanceMetadataRequest(allComputeFieldsUrl);
+            
+            string[] fields = allFieldsResponse?.Split(Environment.NewLine.ToCharArray(), StringSplitOptions.RemoveEmptyEntries);
+
+            if (fields == null || fields.Count() <= 0)
+            {
+                CoreEventSource.Log.CannotObtainAzureInstanceMetadata();
+            }
+
+            return fields;
+        }
+
+        /// <summary>
+        /// Gets the value of a specific field from the imds link asynchronously, and returns it.
+        /// </summary>
+        /// <param name="baseUrl">Url of the Azure Instance Metadata service</param>
+        /// <param name="fieldName">Specific imds field to retrieve a value for</param>
+        /// <param name="apiVersion">rest Api version to append to the constructed Uri</param>
+        /// <param name="textFormatArg">query-string argument to request text data be returned</param>
+        /// <returns>an array of field names available, or null</returns>
+        private static async Task<string> GetAzureInstanceMetadaValue(string baseUrl, string fieldName, string apiVersion, string textFormatArg)
+        { 
+            string metadataFieldUrl = $"{baseUrl}/{fieldName}?{textFormatArg}&{apiVersion}";
+
+            string fieldValue = await MakeAzureInstanceMetadataRequest(metadataFieldUrl);
+
+            return fieldValue;
+        }
+
+        private static async Task<string> MakeAzureInstanceMetadataRequest(string metadataRequestUrl)
+        {
+            string requestResult = string.Empty;
+
+            SdkInternalOperationsMonitor.Enter();
+            try
+            {
+#if NETSTANDARD1_3
+
+                using (var getFieldValueClient = new HttpClient())
+                {
+                    getFieldValueClient.DefaultRequestHeaders.Add("Metadata", "True");
+                    requestResult = await getFieldValueClient.GetStringAsync(metadataRequestUrl).ConfigureAwait(false);
+                }
+
+#elif NET45 || NET46
+
+                WebRequest request = WebRequest.Create(metadataRequestUrl);
+                request.Method = "POST";
+                // request.ContentType = "text";
+                // request.Headers[ContentEncodingHeader] = "utf-8";
+                request.Headers.Add("Metadata", "true");
+                using (WebResponse response = await request.GetResponseAsync().ConfigureAwait(false))
+                {
+                    if (response is HttpWebResponse httpResponse && httpResponse.StatusCode == HttpStatusCode.OK)
+                    {
+                        StreamReader content = new StreamReader(httpResponse.GetResponseStream());
+                        {
+                            requestResult = content.ReadToEnd();
+                        }
+                    }
+                }
+
+#else
+#error Unknown framework
+#endif
+            }
+            catch (AggregateException ex)
```

Comment:
What if we get different exception?

Reply:
Changed to capture all exceptions and log them. This call can fail without any major consequence. #Closed